custom_tables.md

```diff
@@ -0,0 +1,140 @@
+
+An example:
+
+Given 1:1 table `person_additional_infos`
+
+| Column             | Descripton                                              |
+| ------------------ | ------------------------------------------------------- |
+| `id`               | Integer, PK                                             |
+| `person_id`        | Integer, FK on table `personen`                         |
+| `nickname`         | String                                                  |
+| `lucky_number`     | Integer                                                 |
+| `favourite_ide_cd` | String, represents a code in code table `favourite_ide` |
+
+and 1:n table `person_hobbies_n`
+
+| Column      | Descripton                      |
+| ----------- | ------------------------------- |
+| `id`        | Integer, PK                     |
+| `person_id` | Integer, FK on table `personen` |
+| `name`      |                                 |
+
+the following `Masterdata.Person.Update` message
+
+```json
+{
+    "eventType":"Masterdata.Person.Update",
+    "data": {
+        "personReference":"123456",
+        "firstName":"Fred",
+        "customTables": {
+            "person_additional_infos": {
+                "nickname": "Fredu",
+                "lucky_number": 13,
+                "favourite_ide_cd": "VIM"
+            },
+            "person_hobbies_n": [
+                {
+                    "id": 1,
+                    "name": "Knitting"
+                },
+                {
+                    "name": "Running"
+                },
+                {
+                    "id": 3,
+                    "_delete": true
+                }
+            ]
+        }
+    }
+}
+```
+
+* will change the first name of the person with reference `123456` to "Fred" (standard behaviour of `Masterdata.Person.Update`)
+* will manage the contents of the custom tables as specified (additional custom table data behaviour)
+  * the FK `person_id` of the rows custom tables will automatically be set to the id of the person with reference `123456`
+
+### Supported messages
+
+Custom data can be sent alongside any message/event type that is clearly related to a business entity. For now we save us the trouble to maintain an exhaustive list of supported messages. The system actively rejects messages and tells you via error if custom table data cannnot be sent alongside your message type.
+
+### Update behaviour
+
+For 1:1 tables
+
+* If no record is present it will be created.
+* If there already is a record, it will be updated with the fields you provide.
+* To delete an existing record, pass `{ "_delete": true }`
+  * This operation is idempotent, it will not fail if no record is present.
+  * `_delete` is not supported as a table column name.
+
+For 1:n tables, each array entry is handled as follows
+
+* If no id is present the record will be created.
+* If an id is present, the record will be updated with the fields you provide.
```

Comment:
will there be an error if the id does not exist?

Reply:
yes, we delegate to the custom model commands and return all of their validations